clients/src/main/java/org/apache/kafka/clients/admin/CreateTopicsResult.java

```diff
@@ -68,6 +69,19 @@ protected CreateTopicsResult(Map<String, KafkaFuture<TopicMetadataAndConfig>> fu
         return futures.get(topic).thenApply(TopicMetadataAndConfig::config);
     }
 
+    /**
+     * Returns a future that provides topic ID for the topic when the request completes.
+     * <p>
+     * If broker version doesn't support replication factor in the response, throw
+     * {@link org.apache.kafka.common.errors.UnsupportedVersionException}.
+     * If broker returned an error for topic configs, throw appropriate exception. For example,
+     * {@link org.apache.kafka.common.errors.TopicAuthorizationException} is thrown if user does not
+     * have permission to describe topic configs.
+     */
+    public KafkaFuture<Uuid> topicId(String topic) {
+        return futures.get(topic).thenApply(TopicMetadataAndConfig::topicId);
+    }
+    
     /**
      * Returns a future that provides number of partitions in the topic when the request completes.
```

Comment:
I noticed all these comments say "If broker version doesn't support **replication factor** in the response..." should each actually say whatever the method is returning (numPartitions, topicId, etc.)

Reply:
I think we could either do that or just return Uuid.ZERO_UUID

Reply:
Returning `Uuid.ZERO_UUID` should be fine. The comment just seems odd, I am not sure why we have it that way in the other methods.

Reply:
I thought it was odd too.